### PR TITLE
Add research tool dependencies and checks 

### PR DESCRIPTION
**Pull request by hechtlC**
Add dependencies for the different research tools and distributions.
Add functionality to check if all required dependencies of a research tool are installed.
Add functionality to check the current linux distribution.

resolves se-sic/VaRA#626
resolves se-sic/VaRA#740

### CODE REVIEW

File: varats/varats/tools/research_tools/vara.py [hunk 1]
```diff
@@ -138,7 +138,24 @@ class VaRA(ResearchTool[VaRACodeBase]):
...
 
-    __DEPENDENCIES = Dependencies({Distro.DEBIAN: ["libboost-all-dev"]})
+    __DEPENDENCIES = Dependencies({
+        Distro.DEBIAN: [
+            "libboost-all-dev", "zlib1g-dev", "sqlite3", "libsqlite3-dev",
+            "bear", "python3", "doxygen", "graphviz", "python3-pip", "libxml2",
+            "libxml2-dev", "libncurses5-dev", "libncursesw5-dev", "swig",
+            "build-essential", "g++", "cmake", "libz3-dev", "libedit-dev",
+            "python3-sphinx", "libomp-dev", "libcurl4-openssl-dev",
+            "ninja-build"
```
Comment (vulder) [inline varats/varats/tools/research_tools/vara.py]: Please shorten the list with only these deps:
libsqlite3-dev libcurl4-openssl-dev libboost-all-dev libpapi-dev googletest libxml2-dev

File: varats/varats/tools/driver_build_setup.py [hunk 2]
```diff
@@ -250,6 +251,15 @@ def __build_setup_init(
...
         source_location.mkdir(parents=True)
 
+    distro = Distro.get_current_distro()
+    if distro:
+        if not tool.get_dependencies().has_dependencies_for_distro(distro):
+            missing_deps = tool.get_dependencies(
+            ).check_dependencies_for_distro(distro)
+            print(
+                f"The following dependencies have to be installed: {missing_deps}"
+            )
```
Comment (vulder) [inline varats/varats/tools/driver_build_setup.py:262]:
```suggestion
            )
            return
```

File: varats/varats/tools/research_tools/research_tool.py [hunk 3]
```diff
@@ -72,6 +88,36 @@ def has_dependencies_for_distro(self, distro: Distro) -> bool:
...
+    def check_dependencies_for_distro(self, distro: Distro) -> tp.List[str]:
+        """
+        Given a distro, check if all specified dependencies are installed
+        and return the list of not installed dependencies.
+        Args:
+            distro: the distro to use
+
+        Returns:
+            a list containing all not installed dependencies
+        """
```
Comment (vulder) [inline varats/varats/tools/research_tools/research_tool.py]:
```suggestion
    def get_missing_dependencies_for_distro(self, distro: Distro) -> tp.List[str]:
        """
        Given a distro, return all not installed dependencies.

        Args:
            distro: to use

        Returns:
            a list containing all not installed dependencies
        """
```

File: varats/varats/tools/research_tools/vara.py [hunk 4]
```diff
@@ -146,7 +146,18 @@ class VaRA(ResearchTool[VaRACodeBase]):
...
+    __DEPENDENCIES = Dependencies({
+        Distro.DEBIAN: [
+            "libboost-all-dev", "libpapi-dev", "googletest", "libsqlite3-dev",
+            "libxml2-dev", "libcurl4-openssl-dev"
+        ],
+        Distro.ARCH: [
+            "boost-libs", "boost", "which", "zlib", "sqlite3", "ncurses",
+            "make", "python3", "doxygen", "libxml2", "swig", "gcc", "cmake",
+            "z3", "libedit", "graphviz", "python-sphinx", "openmp", "curl",
+            "python-pip"
```
Comment (vulder) [inline varats/varats/tools/research_tools/vara.py]:
```suggestion
            "boost-libs", "boost", "sqlite3", "libxml2", "cmake", "curl"
```

**Review by vulder — APPROVED** [review]

**Review by boehmseb — APPROVED** [review]

**Pull request merged**